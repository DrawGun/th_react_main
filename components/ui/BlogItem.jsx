class BlogItem extends React.Component {
  constructor(props) {
    super(props);

    this._incrementLikes = this.props.incrementLikes.bind(this);
  }

  render() {
    const { id, text, image, meta } = this.props;
    const { src, alt, width, height } = image;
    const { author, createdAt, updatedAt, likes } = meta;

    return (
      <div>
        <TextBox>
          {text}
        </TextBox>
        <Image
          src={src}
          width={width}
          height={height}
          alt={alt} />
        <PostMetaData
          author={author}
          createdAt={createdAt}
          updatedAt={updatedAt} />

        <Like
          id={id}
          likes={likes}
          incrementLikes={this._incrementLikes} />

      </div>
    )
  }

};

BlogItem.propTypes = {
  text: React.PropTypes.string,
  image: React.PropTypes.shape(Image.propTypes),
  meta: React.PropTypes.object,
  incrementLikes: React.PropTypes.func
};

BlogItem.defaultProps = {
  text: "Some text",
  image: {},
  meta: {}
};
