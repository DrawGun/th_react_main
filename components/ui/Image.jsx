class Image extends React.Component {

  render() {
    const {
      src,
      width,
      height,
      alt
    } = this.props;

    return (
      <div>
        <img
          src={src}
          width={width}
          height={height}
          alt={alt} />
      </div>
    )
  }

}

Image.propTypes = {
  src: React.PropTypes.string,
  width: React.PropTypes.string,
  height: React.PropTypes.string,
  alt: React.PropTypes.string
};

Image.defaultProps = {
  src: "https://facebook.github.io/react/img/logo.svg",
  width: "100px",
  height: "100px",
  alt: "Alt"
};
