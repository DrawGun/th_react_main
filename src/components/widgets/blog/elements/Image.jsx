import React from 'react';
import PropTypes from 'prop-types';

class Image extends React.Component {
  render() {
    const {
      src,
      width,
      height,
      alt
    } = this.props;

    return (
      <div>
        <img
          src={src}
          width={width}
          height={height}
          alt={alt} />
      </div>
    );
  }
}

Image.propTypes = {
  src: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  alt: PropTypes.string
};

Image.defaultProps = {
  src: 'https://facebook.github.io/react/img/logo.svg',
  width: '100px',
  height: '100px',
  alt: 'Alt'
};

export default Image;
