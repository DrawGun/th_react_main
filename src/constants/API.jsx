export const API_ROOT = 'http://localhost:3001';
export const APP_ROOT = 'http://localhost:3000';
