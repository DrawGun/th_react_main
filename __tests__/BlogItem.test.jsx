import React from 'react';
import ReactDOM from 'react-dom';

import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

import BlogItem from '../src/components/widgets/blog/Item';
import Image from '../src/components/widgets/blog/elements/Image';

import history from 'helpers/routes/history';

import { fakeStore } from 'helpers/fakeStore';
import { fakeState } from 'constants/fakeState';

describe('BlogItem', () => {
  const state = fakeState;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={fakeStore(state)}>
        <Router history={history}>
          <BlogItem />
        </Router>
      </Provider>, div);
  });

  describe('BlogItem', () => {
    it('should render the Image', () => {
      const itemProps = {
        src: '/img/5',
        width: '1px',
        height: '1px',
        alt: 'Alt'
      };

      const item = shallow(<BlogItem image={itemProps} />);

      const header = <Image
        src='/img/5'
        width='1px'
        height='1px'
        alt='Alt' />;

      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'Hello, World!',
        id: 5,
        image: {
          alt: 'Hello, World',
          src: '/images/World.jpg'
        }
      };

      const item = shallow(<BlogItem post={itemProps} />);

      expect(item).toMatchSnapshot();
    });
  });
});
