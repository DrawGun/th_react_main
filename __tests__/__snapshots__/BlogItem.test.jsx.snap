// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem BlogItem should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    header={
        <Title
            text="Some text"
            url="Some url"
        />
    }
>
    <Media
        bsClass="media"
        componentClass="div"
    >
        <MediaLeft
            align="top"
            bsClass="media-left"
        >
            <Image
                alt="Alt"
                height="100px"
                src="https://facebook.github.io/react/img/logo.svg"
                width="100px"
            />
        </MediaLeft>
        <MediaBody
            bsClass="media-body"
            componentClass="div"
        >
            <MetaData
                author="Author"
                createdAt="19.08.2017"
                updatedAt="19.08.2017"
            />
            <Connect(Likes)
                itemId={1}
                itemType="Post"
            />
            <Link
                linkClassNamse="link"
                to="Some editUrl"
                wrapperClassNames="edit-post-link"
            >
                Edit
            </Link>
        </MediaBody>
    </Media>
</Panel>,
  "nodes": Array [
    <Panel
      bsClass="panel"
      bsStyle="default"
      defaultExpanded={false}
      header={
            <Title
                  text="Some text"
                  url="Some url"
            />
      }
>
      <Media
            bsClass="media"
            componentClass="div"
      >
            <MediaLeft
                  align="top"
                  bsClass="media-left"
            >
                  <Image
                        alt="Alt"
                        height="100px"
                        src="https://facebook.github.io/react/img/logo.svg"
                        width="100px"
                  />
            </MediaLeft>
            <MediaBody
                  bsClass="media-body"
                  componentClass="div"
            >
                  <MetaData
                        author="Author"
                        createdAt="19.08.2017"
                        updatedAt="19.08.2017"
                  />
                  <Connect(Likes)
                        itemId={1}
                        itemType="Post"
                  />
                  <Link
                        linkClassNamse="link"
                        to="Some editUrl"
                        wrapperClassNames="edit-post-link"
                  >
                        Edit
                  </Link>
            </MediaBody>
      </Media>
</Panel>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <BlogItem
        editUrl="Some editUrl"
        id={1}
        image={Object {}}
        meta={Object {}}
        post={
                Object {
                        "id": 5,
                        "image": Object {
                          "alt": "Hello, World",
                          "src": "/images/World.jpg",
                        },
                        "title": "Hello, World!",
                      }
        }
        title="Some text"
        url="Some url"
/>,
      "_debugID": 52,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BlogItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "editUrl": "Some editUrl",
          "id": 1,
          "image": Object {},
          "meta": Object {},
          "post": Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          },
          "title": "Some text",
          "url": "Some url",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Panel
          bsClass="panel"
          bsStyle="default"
          defaultExpanded={false}
          header={
                    <Title
                              text="Some text"
                              url="Some url"
                    />
          }
>
          <Media
                    bsClass="media"
                    componentClass="div"
          >
                    <MediaLeft
                              align="top"
                              bsClass="media-left"
                    >
                              <Image
                                        alt="Alt"
                                        height="100px"
                                        src="https://facebook.github.io/react/img/logo.svg"
                                        width="100px"
                              />
                    </MediaLeft>
                    <MediaBody
                              bsClass="media-body"
                              componentClass="div"
                    >
                              <MetaData
                                        author="Author"
                                        createdAt="19.08.2017"
                                        updatedAt="19.08.2017"
                              />
                              <Connect(Likes)
                                        itemId={1}
                                        itemType="Post"
                              />
                              <Link
                                        linkClassNamse="link"
                                        to="Some editUrl"
                                        wrapperClassNames="edit-post-link"
                              >
                                        Edit
                              </Link>
                    </MediaBody>
          </Media>
</Panel>,
        "_debugID": 53,
        "_renderedOutput": <Panel
          bsClass="panel"
          bsStyle="default"
          defaultExpanded={false}
          header={
                    <Title
                              text="Some text"
                              url="Some url"
                    />
          }
>
          <Media
                    bsClass="media"
                    componentClass="div"
          >
                    <MediaLeft
                              align="top"
                              bsClass="media-left"
                    >
                              <Image
                                        alt="Alt"
                                        height="100px"
                                        src="https://facebook.github.io/react/img/logo.svg"
                                        width="100px"
                              />
                    </MediaLeft>
                    <MediaBody
                              bsClass="media-body"
                              componentClass="div"
                    >
                              <MetaData
                                        author="Author"
                                        createdAt="19.08.2017"
                                        updatedAt="19.08.2017"
                              />
                              <Connect(Likes)
                                        itemId={1}
                                        itemType="Post"
                              />
                              <Link
                                        linkClassNamse="link"
                                        to="Some editUrl"
                                        wrapperClassNames="edit-post-link"
                              >
                                        Edit
                              </Link>
                    </MediaBody>
          </Media>
</Panel>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    editUrl="Some editUrl"
    id={1}
    image={Object {}}
    meta={Object {}}
    post={
        Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          }
    }
    title="Some text"
    url="Some url"
/>,
}
`;
